src/connector/src/parser/common.rs

```diff
@@ -18,19 +18,28 @@ use risingwave_common::array::{ListValue, StructValue};
 use risingwave_common::types::{DataType, Datum, Decimal, ScalarImpl};
 use risingwave_common::util::iter_util::ZipEqFast;
 use risingwave_expr::vector_op::cast::{
-    i64_to_timestamp, i64_to_timestamptz, str_to_date, str_to_time, str_to_timestamp,
-    str_with_time_zone_to_timestamptz,
+    i32_to_date, i64_to_time, i64_to_timestamp, i64_to_timestamptz, str_to_date, str_to_time,
+    str_to_timestamp, str_with_time_zone_to_timestamptz,
 };
 use simd_json::value::StaticNode;
 use simd_json::{BorrowedValue, ValueAccess};
 
-use crate::{ensure_int, ensure_str};
+use crate::{ensure_i32, ensure_int, ensure_str, simd_json_ensure_float};
 
 fn do_parse_simd_json_value(dtype: &DataType, v: &BorrowedValue<'_>) -> Result<ScalarImpl> {
-    use crate::simd_json_ensure_float;
-
     let v = match dtype {
-        DataType::Boolean => v.as_bool().ok_or_else(|| anyhow!("expect bool"))?.into(),
+        DataType::Boolean => match v {
+            BorrowedValue::Static(StaticNode::Bool(_)) => {
+                v.as_bool().ok_or_else(|| anyhow!("expect bool"))?.into()
+            }
+            // debezium converts bool to int, false -> 0, true -> 1, for mysql and postgres
+            BorrowedValue::Static(StaticNode::I64(i)) => match i {
+                0i64 => ScalarImpl::Bool(false),
+                1i64 => ScalarImpl::Bool(true),
+                _ => anyhow::bail!("expect bool, but found {v}"),
+            },
+            _ => anyhow::bail!("expect bool, but found {v}"),
```

Comment:
* When matching bool, it can be just `BorrowedValue::Static(StaticNode::Bool(b)) => (*b).into()`
* When matching int, avoid relying on `0` or `1` being an `I64` rather than `U64`. Prefer `as_` to cover all cases.